proof/refine/RISCV64/Syscall_R.thy

```diff
@@ -1970,8 +1902,14 @@ lemma chargeBudget_corres:
   apply (rule_tac Q="\<lambda>s. sc_at (cur_sc s) s" in corres_cross_add_abs_guard)
    apply (fastforce intro: cur_sc_tcb_sc_at_cur_sc)
   apply add_cur_tcb'
+  apply (rule corres_stateAssert_ignore, simp)
+  apply clarsimp
+  apply (rule corres_stateAssert_ignore, simp)
+   apply (fastforce intro: sch_act_wf_cross)
   apply (rule corres_underlying_split[rotated 2, OF gets_sp getCurSc_sp])
    apply (corresKsimp corres: getCurSc_corres)
+  apply (rule corres_stateAssert_ignore, simp)
+   apply (fastforce intro: active_sc_at'_cross)
```

Comment:
this looks a bit odd... the first `corres_stateAssert_ignore` looks like it could be combined with the subsequent clarsimp:
`apply (rule corres_stateAssert_ignore; clarsimp)`
the second and third look like that simp isn't doing much, and they could be:
`apply (rule corres_stateAssert_ignore, fastforce intro: sch_act_wf_cross)`
and
`apply (rule corres_stateAssert_ignore, fastforce intro: active_sc_at'_cross)`

Reply:
I was able to tidy up this area of the proof. One of the `clarsimp`s was needed because we just did an `unfolding` that left a `K_bind`, which I don't like, I combined the `fastforce`s as you wrote, and then changed a `corresKsimp` to a `corres`.

Reply:
Cool. Thank you, and the switch to `corres` gives a chance that this will be understandable if it breaks later.